src/pages/workspaces/workspace/ShareWorkspaceModal.js

```diff
@@ -27,14 +27,49 @@ const styles = {
   pending: {
     textTransform: 'uppercase', fontWeight: 500,
     color: colors.orange[0]
-  },
-  roleSelect: base => ({
-    ...base,
-    width: 200,
-    marginTop: '0.25rem'
-  })
+  }
 }
 
+const AclInput = ({ value, onChange, disabled }) => {
+  const { accessLevel, canShare, canCompute } = value
+  return div({ style: { display: 'flex', marginTop: '0.25rem' } }, [
+    div({ style: { width: 200 } }, [
+      h(Select, {
+        isSearchable: false,
+        isDisabled: disabled,
+        getOptionLabel: o => Utils.normalizeLabel(o.value),
+        value: accessLevel,
+        onChange: o => onChange({
+          ...value,
+          accessLevel: o.value,
+          ...Utils.switchCase(o.value,
+            ['READER', () => ({ canCompute: false })],
+            ['WRITER', () => ({ canCompute: true })],
+            ['OWNER', () => ({ canCompute: true, canShare: true })]
+          )
+        }),
+        options: accessLevel === 'PROJECT_OWNER' ? ['PROJECT_OWNER'] : ['READER', 'WRITER', 'OWNER'],
+        menuPortalTarget: document.getElementById('modal-root')
+      })
+    ]),
+    div({ style: { marginLeft: '1rem' } }, [
+      div([
+        h(LabeledCheckbox, {
+          disabled: disabled || accessLevel === 'OWNER',
+          checked: canShare,
+          onChange: () => onChange(_.update('canShare', b => !b, value))
+        }, [' Can share'])
+      ]),
+      div([
+        h(LabeledCheckbox, {
+          disabled: disabled || accessLevel !== 'WRITER',
```

Comment:
Can we make the "canCompute" labeled checkbox look more obviously disabled when you've selected a reader role?

 Mousing over the text still gives a pointer leading me to think you can click it when you can't. I know elsewhere we have a disabled cursor with a little X symbol thing when hovering over disabled clickables which could be helpful. Maybe a coloring/visual indicator too.

Reply:
I agree. Enabled, unselected checkboxes look more disabled to me than disabled, unselected checkboxes. Suggestion: swap the styling of the checkbox itself and dim the label text of disabled checkboxes.

Reply:
Good suggestions, done.